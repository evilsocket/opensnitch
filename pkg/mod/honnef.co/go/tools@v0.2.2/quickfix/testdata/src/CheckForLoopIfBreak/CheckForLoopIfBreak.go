package pkg

func done() bool { return false }

var a, b int
var x bool

func fn() {
	for {
		if done() { // want `could lift into loop condition`
			break
		}
	}

	for {
		if !done() { // want `could lift into loop condition`
			break
		}
	}

	for {
		if a > b || b > a { // want `could lift into loop condition`
			break
		}
	}

	for {
		if x && (a == b) { // want `could lift into loop condition`
			break
		}
	}

	for {
		if done() { // want `could lift into loop condition`
			break
		}
		println()
	}

	for {
		println()
		if done() {
			break
		}
	}

	for {
		if done() {
			println()
			break
		}
	}
}
