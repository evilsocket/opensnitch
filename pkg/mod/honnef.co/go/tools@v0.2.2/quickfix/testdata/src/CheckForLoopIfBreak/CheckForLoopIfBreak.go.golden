package pkg

func done() bool { return false }

var a, b int
var x bool

func fn() {
	for !done() {

	}

	for done() {

	}

	for a <= b && b <= a {

	}

	for !x || !(a == b) {

	}

	for !done() {

		println()
	}

	for {
		println()
		if done() {
			break
		}
	}

	for {
		if done() {
			println()
			break
		}
	}
}
